### PR TITLE
Fix Dependency Loop Error Introduced with #14676

### PR DESCRIPTION
**Pull request by emargolis**
#### Problem
Described in #15121

#### Change overview
Fixed

#### Testing
existing tests

### CODE REVIEW

File: src/include/platform/internal/GenericConfigurationManagerImpl.cpp [hunk 1]
```diff
@@ -335,9 +334,11 @@ CHIP_ERROR GenericConfigurationManagerImpl<ConfigClass>::GetSpake2pIterationCoun
 template <class ConfigClass>
 CHIP_ERROR GenericConfigurationManagerImpl<ConfigClass>::GetSpake2pSalt(uint8_t * buf, size_t bufSize, size_t & saltLen)
 {
-    CHIP_ERROR err                                                   = CHIP_NO_ERROR;
-    char saltB64[BASE64_ENCODED_LEN(chip::kPBKDFMaximumSaltLen) + 1] = { 0 };
-    size_t saltB64Len                                                = 0;
+    // The maximum length of raw salt is chip::kPBKDFMaximumSaltLen (32 bytes).
+    static constexpr size_t kSpake2pSalt_MaxBase64Len = BASE64_ENCODED_LEN(32) + 1;
```
Comment (andy31415) [inline src/include/platform/internal/GenericConfigurationManagerImpl.cpp:338]: The constants should be taken into some other place that is visible more generally or have a different header with different include rules.

Hardcoding these will make things much harder to maintain.

**Review by andy31415 — APPROVED** [review]
Approve as a fix for flaky builds, however for long term maintainability more work is required' Hardcoding things like `32` instead of using constants because it is hard to include things should be solvable by having separate includes and dependencies.

File: src/include/platform/internal/GenericConfigurationManagerImpl.cpp (hunk 1 above)
Comment (bzbarsky-apple) [inline src/include/platform/internal/GenericConfigurationManagerImpl.cpp:338]: Can we put this constant somewhere in the crypto headers?  Can everything involved include those?

Comment (emargolis) [reply src/include/platform/internal/GenericConfigurationManagerImpl.cpp]: I've updated the fix and moved some of the Spake2p defines to the CHIPCryptoPAL.h

Comment (emargolis) [reply src/include/platform/internal/GenericConfigurationManagerImpl.cpp]: I've updated the fix and moved some of the Spake2p defines to the CHIPCryptoPAL.h

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**